assets/tests/hooks/useDetour.test.ts

```diff
@@ -179,12 +190,37 @@ describe("useDetour", () => {
   })
 
   test("when `endPoint` is set, `canUndo` is `false`", async () => {
-    const { result } = renderHook(useDetour)
+    const { result } = renderHook(() => useDetour("routePatternId"))
+
+    jest
+      .mocked(fetchDetourMissedStops)
+      .mockResolvedValue(stopFactory.buildList(3))
 
     act(() => result.current.addConnectionPoint({ lat: 0, lon: 0 }))
     act(() => result.current.addConnectionPoint({ lat: 0, lon: 0 }))
 
+    await waitFor(() => {
```

Comment:
Is this (and the mock-resolved-value above) necessary as part of this test? It almost looks like setup meant for a different set of assertions?

Reply:
I just added it to make the test more realistic since the hook is now doing that API logic when an endpoint is added, but I could be convinced to remove it.

Reply:
I'd vote in favor of removing it, since I think it makes it a bit more confusing to tell exactly _when_ the bits that the test asserts on happen, and makes it seem like the test relies on this setup.

Reply:
Ahh, I remember now - without this, the API logic still fires with the default mock resolved value of `null` and then triggers a `Warning: An update to TestComponent inside a test was not wrapped in act(...).` error. That said, to actually make sure I'm avoiding a race condition I should update the `waitFor` to look for `.no.toBeUndefined()`.

Reply:
6f9e27d1

Reply:
That's actually... kind of surprising, since the update is triggered by `act(() => result.current.addConnectionPoint({ lat: 0, lon: 0 }))`, which _is_ wrapped in `act`.

But I'll let this go. It's annoying, but not a huge deal, and we can take a more holistic approach to cleaning these up later if we decide that's a thing we should do.

Reply:
That's what triggers the `useEffect` to fire, but the promise from `fetchDetourMissedStops` still has to resolve, hence it happening outside of that `act`.

Reply:
Oooooh - that's annoying! I wish there were a way to make promises and things like that in tests resolve synchronously.